src/coreclr/vm/jitinterface.cpp

```diff
@@ -11975,7 +11975,6 @@ bool CEEInfo::getReadonlyStaticFieldValue(CORINFO_FIELD_HANDLE fieldHnd, uint8_t
 
```

Comment:
Add `_ASSERT(valueOffset >= 0);` to the batch of asserts above?

Reply:
I decided to do a fault-free bound check on VM side because:
1) User might use negative index to access RVA
2) JIT doesn't know RVA size

so jit just relies on `getReadonlyStaticFieldValue` to do bound checking and return true/false, in case of false jit won't try to optimize anything and will let the codegen to throw an exception (if it hits)

Reply:
Ah, actually for valueOffset it's better to ensure it's not negative. so ignore my previous comment